providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/EventStreamObserver.java

```diff
@@ -0,0 +1,101 @@
+package dev.openfeature.contrib.providers.flagd;
+
+import java.util.Map;
+import io.grpc.stub.StreamObserver;
+import dev.openfeature.flagd.grpc.Schema.EventStreamResponse;
+import dev.openfeature.sdk.ProviderEvaluation;
+import com.google.protobuf.Value;
+
+/**
+ * EventStreamObserver handles events emitted by flagd.
+ */
+public class EventStreamObserver implements StreamObserver<EventStreamResponse> {
+    private EventStreamCallback callback;
+    private Boolean cacheEnabled;
+    private Map<String, ProviderEvaluation<Boolean>> booleanCache;
+    private Map<String, ProviderEvaluation<String>> stringCache;
+    private Map<String, ProviderEvaluation<Double>> doubleCache;
+    private Map<String, ProviderEvaluation<Integer>> integerCache;
+    private Map<String, ProviderEvaluation<dev.openfeature.sdk.Value>> objectCache;
+
+
+    static final String CONFIGURATION_CHANGE = "configuration_change";
+    static final String PROVIDER_READY = "provider_ready";
```

Comment:
Both could be private if the intention is to use as constants in this class

Reply:
Done